web3/contract.py

```diff
@@ -1350,7 +1350,7 @@ def __getattr__(self, function_name: str) -> ContractFunction:
             return super().__getattribute__(function_name)
 
     def __call__(
-        self, transaction: TxDict=None, block_identifier: BlockIdentifier='latest'
+        self, transaction: TxParams=None, block_identifier: BlockIdentifier='latest'
```

Comment:
I just saw that python 3.8 introduces [`typing.Literal`](https://docs.python.org/3/library/typing.html#typing.Literal) which will be a nice improvement to the `BlockIdentifier` and other *magic* string based values.  Seems like the kind of thing that we may get a backport available for as well.